wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/MachineSourceImpl.java

```diff
@@ -25,15 +27,28 @@
 
     private String type;
     private String location;
+    private String content;
+
+    protected MachineSourceImpl() {
+
+    }
 
     public MachineSourceImpl(String type, String location) {
+        this(type, location, null);
+    }
+
+    public MachineSourceImpl(String type, String location, String content) {
```

Comment:
Don't get why we need this constructor as location and content are mutually exclusive